cep-XXXX.md

```diff
@@ -0,0 +1,216 @@
+
+- Distributable package names: represented by a concrete, downloadable, extractable conda artifact.
+- Virtual package names: not backed by any physical artifact. They only exist on the client side.
+
+#### Package names
+
+A distributable package name MUST only consist of lowercase ASCII letters, numbers, hyphens, periods
+and underscores. It MUST start with a letter, a number, or a single underscore. It MUST NOT include
+two consecutive separators (hyphen, period, underscore).
+
+Virtual package names MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and
+underscores. They MUST NOT use two consecutive separators, with one exception: they MUST start with
+two underscores.
+
+This means that distributable package names MUST match the following case-insensitive
+regex: `^(([a-z0-9])|([a-z0-9_](?!_)))[._-]?([a-z0-9]+(\.|-|_|$))*$`.
+
+Virtual package names MUST follow this regex: `^__[a-z0-9][._-]?([a-z0-9]+(\.|-|_|$))*$`.
+
+In all cases, the maximum length of a package name MUST NOT exceed 64 characters.
+
+#### Version strings
+
+Version strings MUST only consist of digits, periods, lowercase ASCII letters, underscores, plus
+symbols, and exclamation marks. Additional rules apply but are out of scope in this CEP and will be
+discussed separately.
+
+The maximum length of a version string MUST NOT exceed 64 characters.
+
+#### Build strings
+
+Builds strings MUST only consist of ASCII letters, numbers, periods, plus symbols, and underscores.
+They MUST match this regex `^[a-zA-Z0-9_\.+]+$`.
+
+The maximum length of a build string MUST NOT exceed 64 characters.
+
+#### Artifact extensions
+
+Artifact extensions MUST only consist of lowercase ASCII letters, numbers and periods. They must start
+and end with a letter or a number. They MUST NOT include two consecutive periods. They MUST match this
+regex `^[a-z0-9](\.?[a-z0-9])*$`.
+
+The maximum length of a file extension MUST NOT exceed 16 characters.
+
+> The conda ecosystem currently recognizes two artifact extensions: `tar.bz2` and `conda`, versioned
+`v1` and `v2` respectively.
+
+#### Filenames
+
+Distributable conda artifacts MUST have a filename following this scheme:
+
+```text
+<package name>-<version string>-<build string>.<extension>
+```
+
+The maximum length of a filename MUST NOT exceed 192 characters.
+
+Virtual conda packages do not exist on disk and SHOULD NOT need filename standardization.
+
+#### Distribution strings
+
+A "distribution string" MAY be used to identify a package artifact without specifying the extension
+or the channel. It MUST match the following syntax:
+
+```text
+<package name>-<version string>-<build string>
+```
+
+Distribution strings apply to both distributable and virtual packages. They are used as the name of the
+directories where artifacts are extracted in the package cache, for example.
+
+> Note: Despite the similarity, distribution strings are not `MatchSpec`-like specifiers and
+> MUST NOT be used as such.
```

Comment:
This would be best moved _before_ filenames as the latter inherit from the `Distribution strings` spec